### ISSUE TITLE
KeyError in writer.generate_repr()

### ISSUE THREAD

**Issue by brigreens** [cclib/cclib#1155]
Hi, 
I am trying to parse an Orca output file and get the optimized coordinates in the form of an xyz file (or even better, the SMILES). 
I have the following code:
```
data = cclib.io.ccread(file_path)
writer = cclib.io.xyzwriter.XYZ(data)
new_repr = writer.generate_repr()
```

The line for writer.generate_repr() throws the following error: `KeyError: 'comments'`

Any help would be appreciated

**Comment by shivupa** [comment]
Hi, Thanks for reporting this!

 I tried this quickly with a [test file from the test suite](https://github.com/cclib/cclib/blob/master/data/ORCA/basicORCA5.0/dvb_gopt.out). `new_repr` was generated correctly in this case.

Do you have a example Orca output file that causes this that you could provide? Also what versions of Orca and cclib are you using?

**Comment by brigreens** [comment]
I attached my output file. When I tried running the above code, it was compressed with bzip2, but github won't let me attach that file type so I saved it as a text file. Maybe you can compress it back. I am using Orca 4.2.0 and cclib 1.7.2

[8-45_AB3.out.txt](https://github.com/cclib/cclib/files/9630150/8-45_AB3.out.txt)

**Comment by shivupa** [comment]
Hm. I wasn't able to reproduce this error. 

```
Python 3.10.7 (main, Sep  6 2022, 21:22:27) [GCC 12.2.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.5.0 -- An enhanced Interactive Python. Type '?' for help.
In [1]: import cclib
In [2]: data = cclib.io.ccread("8-45_AB3.out.txt.bz2")
In [3]: writer = cclib.io.xyzwriter.XYZ(data)
In [4]: new_repr = writer.generate_repr()
In [5]: cclib.__version__
Out[5]: '1.7.2'
In [6]: new_repr
Out[6]: '53\n[Geometry 95]\nC     -0.7674940000   -0.6969810000   -0.0338640000\nS      0.0788980000    0.4533590000   -1.0036060000\nC      1.2596970000    0.7105570000    0.2862550000\nC      0.9738760000   -0.1210980000    1.3787590000\nC     -0.1896610000   -0.8994050000    1.1824900000\nN      1.7002460000   -0.1360570000    2.5577610000\nC      2.2775550000    1.6674050000    0.1372670000\nS      3.3339610000    2.1811920000    1.4331550000\nC      4.2616660000    3.2826410000    0.4470960000\nC      3.6978330000    3.2842210000   -0.8578360000\nC      2.5428740000    2.3639920000   -1.0882890000\nO      1.9314420000    2.2399280000   -2.1523590000\nN      4.2826390000    4.6086290000   -2.8083750000\nN      4.0771790000    4.0336390000   -1.8513900000\nC      5.3306630000    4.0042730000    0.9626630000\nS      5.8078410000    3.8187530000    2.6548620000\nC      7.1333520000    4.9626900000    2.4457300000\nC      7.2007500000    5.4013760000    1.0981810000\nC      6.1951460000    4.8805360000    0.3000090000\nN      8.1442840000    6.3205720000    0.6648270000\nC      7.9205780000    5.3386060000    3.5253250000\nS      9.1235730000    6.6193550000    3.5049730000\nC      9.6319160000    6.3608810000    5.1595350000\nC      8.8200230000    5.3412550000    5.7242900000\nC      7.8110240000    4.7221310000    4.8236820000\nO      7.0233980000    3.8278870000    5.1349160000\nN      8.8337830000    4.4000880000    7.9669070000\nN      8.8907380000    4.8912270000    6.9458350000\nC     10.6439250000    7.1149730000    5.7426690000\nS     11.5955680000    8.2353730000    4.7676080000\nC     12.5521990000    8.7309310000    6.1573300000\nC     12.1441160000    8.0260750000    7.3146650000\nC     11.1011110000    7.1292440000    7.0573600000\nN     12.6274690000    8.2819730000    8.5687350000\nC     13.5179110000    9.7412300000    6.0051890000\nS     13.4571550000   10.8266800000    4.6295040000\nC     14.9227580000   11.6122780000    5.0041100000\nC     15.4456720000   11.0998180000    6.1872140000\nC     14.6864570000    9.9713310000    6.8114000000\nO     15.0671090000    9.3633420000    7.8232590000\nN     17.4198730000   11.8170910000    7.4336410000\nN     16.5238350000   11.5361750000    6.7971750000\nH     -1.6413690000   -1.1853100000   -0.4300720000\nH     -0.5673330000   -1.5944470000    1.9190770000\nH      2.7002660000   -0.0467680000    2.4614120000\nH      1.4683910000   -0.9006740000    3.1703310000\nH      6.1177220000    5.1174700000   -0.7493520000\nH      9.0852360000    6.1516450000    0.9882190000\nH      8.1375370000    6.4677960000   -0.3316090000\nH     10.6560390000    6.5424780000    7.8450150000\nH     13.5829130000    8.6336060000    8.5884240000\nH     12.4482010000    7.5616330000    9.2477140000\nH     15.3110590000   12.4035050000    4.3916830000\n'
```

Was this set of calls part of some larger script?

**Comment by brigreens** [comment]
Interesting. Yes it is but even if I put it in a new script, I still get the issue. I am running it on Python 3.7.5

**Comment by brigreens** [comment]
This is the exact error message:
```
KeyError                                  Traceback (most recent call last)
<ipython-input-7-1ffb75a0f70a> in <module>
      1 data = cclib.io.ccread('B973C_hexamers/neutral/8-45_AB3.out.bz2')
      2 writer = cclib.io.xyzwriter.XYZ(data)
----> 3 new_repr = writer.generate_repr()

~/.local/lib/python3.7/site-packages/cclib/io/xyzwriter.py in generate_repr(self)
     71             indices = [-1]
     72         for i in indices:
---> 73             xyzblock.append(self._xyz_from_ccdata(i))
     74 
     75         # Ensure an extra newline at the very end.

~/.local/lib/python3.7/site-packages/cclib/io/xyzwriter.py in _xyz_from_ccdata(self, index)
     82 
     83         atomcoords = self.ccdata.atomcoords[index]
---> 84         existing_comment = "" if not self.ccdata.metadata["comments"] \
     85             else self.ccdata.metadata["comments"][index]
     86 

KeyError: 'comments'
```

**Comment by brigreens** [comment]
Hi thank you so much for your help. You were correct, for some reason in the command line it was saying version 1.7.2 but in my jupyter notebook it was 1.6.1. I fixed the path issue and with version 1.7.2 I no longer have the error.
